src/Syncs/InterCPISync.cpp

```diff
@@ -42,8 +39,7 @@ InterCPISync::InterCPISync(long m_bar, long bits, int epsilon, int partition,boo
 
 	treeNode = nullptr;
         // TODO: Novak's quick fix.
-        // useExisting=false;
-        useExisting=true;
+        useExisting=false;
```

Comment:
why?

Reply:
As far as I can understand, `useExisiting` is used to specify that the communication socket should not be opened/closed because this has been done prior to using the sync. In theory this could be used to run multiple syncs with the same connection without needing to reopen it.

There are multiple issues with this however:
- Nonconformity to other syncs: The flag defaults to `true`, which indicates to use an existing connection. This differs from all other sync protocols and is not taken into account in the unit test for it, which makes it fail.
- Inconsistent server/client behavior: Only in `SyncClient` is the connection no opened if `useExisting=true`. In `SyncServer` the connection is opened regardless. Both sides of the sync, however, do not close the connection if the flag is set.
- Non-configurable: The flag is marked as protected and there is no way to change it from outside of the sync implementation (unless you would create a subclass just to change the flag).

In my opinion, the flag should either be removed entirely or properly implemented. Defaulting it to `false` is only a quick fix.